src/Evolution/Systems/Cce/BoundaryData.hpp

```diff
@@ -0,0 +1,208 @@
+    const Scalar<DataVector>& dt_cartesian_lapse) noexcept;
+/*!
+ * \brief Computes the metric and its first derivative in the intermediate
+ * radial null coordinates
+ *
+ * \details These components are obtained by \cite Barkett2019uae
+ * \cite Bishop1998uk. The most direct comparison is to be made with equation
+ * (31) of \cite Barkett2019uae. Note that the Jacobian factors do not depend on
+ * the time coordinate, so the time derivative of the metric transforms
+ * similarly.
+ */
+void null_metric_and_derivative(
+    gsl::not_null<tnsr::aa<DataVector, 3>*> du_null_metric,
+    gsl::not_null<tnsr::aa<DataVector, 3>*> null_metric,
+    const tnsr::iJ<DataVector, 3>& cartesian_to_spherical_jacobian,
+    const tnsr::aa<DataVector, 3>& dt_psi,
+    const tnsr::aa<DataVector, 3>& psi) noexcept;
+
+/*!
+ * \brief Computes the spatial unit normal vector \f$s^i\f$ to the spherical
+ * worldtube surface and its first time derivative.
+ *
+ * \details Refer to equation (20) of \cite Barkett2019uae for the expression of
+ * the spatial unit normal vector, and equation (23) of \cite Barkett2019uae for
+ * the first time derivative. Refer to \cite Bishop1998uk for more exposition
+ * about the overall construction of the coordinate transformations used for the
+ * intermediate null coordinates.
+ */
+void worldtube_normal_and_derivatives(
+    gsl::not_null<tnsr::I<DataVector, 3>*> worldtube_normal,
+    gsl::not_null<tnsr::I<DataVector, 3>*> dt_worldtube_normal,
+    const Scalar<DataVector>& cos_phi, const Scalar<DataVector>& cos_theta,
+    const tnsr::aa<DataVector, 3>& psi, const tnsr::aa<DataVector, 3>& dt_psi,
+    const Scalar<DataVector>& sin_phi, const Scalar<DataVector>& sin_theta,
+    const tnsr::II<DataVector, 3>& inverse_spatial_metric) noexcept;
+
+/*!
+ * \brief Computes the null 4-vector \f$l^\mu\f$ on the worldtube surface that
+ * is to be used as the CCE hypersurface generator, and the first time
+ * derivative \f$\partial_u l^\mu\f$.
+ *
+ * \details For mathematical description of our choice of the null generator,
+ * refer to equation (22) of \cite Barkett2019uae, and for the first time
+ * derivative see (25) of \cite Barkett2019uae.  Refer to \cite Bishop1998uk for
+ * more exposition about the overall construction of the coordinate
+ * transformations used for the intermediate null coordinates.
+ */
+void null_vector_l_and_derivatives(
+    gsl::not_null<tnsr::A<DataVector, 3>*> du_null_l,
+    gsl::not_null<tnsr::A<DataVector, 3>*> null_l,
+    const tnsr::I<DataVector, 3>& dt_worldtube_normal,
+    const Scalar<DataVector>& dt_lapse, const tnsr::aa<DataVector, 3>& dt_psi,
+    const tnsr::I<DataVector, 3>& dt_shift, const Scalar<DataVector>& lapse,
+    const tnsr::aa<DataVector, 3>& psi, const tnsr::I<DataVector, 3>& shift,
+    const tnsr::I<DataVector, 3>& worldtube_normal) noexcept;
+
+/*!
+ * \brief Computes the partial derivative of the metric and inverse metric in
```

Comment:
Is this suggested for the dox or for the code variable names? I'm happy to add `spacetime` to the variable names, but I'd prefer to refer to metric in the documentation as 'in ... coordinates' because in this context they're actually all the same abstract metric, just evaluated in different coordinate systems.

Reply:
Sorry, we do mean in the dox. Would you be opposed to `of the spacetime and inverse spacetime metric...`?

Reply:
I'm happy putting that in.

Reply:
Okay, great! We've generally found that at least in all the 3+1 split code this eliminates issues of "which metric" (spatial vs. spacetime) and I feel like doing so in the CCE code doesn't hurt unless it makes something technically wrong :)